scanpy/external/pp/_nn_formatted.py

```diff
@@ -0,0 +1,99 @@
+from typing import Optional
+
+import numpy as np
+import warnings
+from scipy.sparse import issparse, coo_matrix, csr_matrix
+from anndata import AnnData
+#from .. import settings
```

Comment:
Is this something we want to add next semester? Maybe add notes to the API spec about it